backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Controllers/FileController.java

```diff
@@ -50,6 +56,9 @@ public ResponseEntity<CustomFile> getFileOrDirectoryById(@PathVariable Long id)
     @GetMapping("/files/directory/root")
     public ResponseEntity<List<CustomFile>> getRootFilesAndDirectories() {
         List<CustomFile> customFiles = fileService.getFilesAtRoot();
+        if (customFiles.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
         return new ResponseEntity<>(customFiles, HttpStatus.OK);
     }
 
```

Comment:
I don't think that `HttpStatus.NOT_FOUND` makes sense here. We did not fail to find something, instead of returned an empty list.